### ISSUE TITLE
Properly turn on/off inspiration fragments when using the next/previous arrows (bug of #384)

### ISSUE THREAD

**Issue by phraenquex** [m2ms/fragalysis-frontend#407]
Fix incomplete implementation of #384.
Spec: turning on/off of the inspiration hits isn't working, this is what had been requested:

1. remember the FULL display settings of the compound/hit above the arrow.

* For "F", if any hits are displayed, consider that as all hits displayed; and if no hits are displayed, then keep that setting.
* (Ideally) if any display settings were customised for either protein, ligand or contacts, then remember those too.
* (pragatically) if the previous point is too difficult, then don't bother.

2. turn off all displayed elements (except protein ribbon)
3. apply the settings of (1) above to the compound below the arrow.

Here's what's not working:  In the snapshot below: I had all inspirations of one compound turned on (PAU-WEI-b9b69149-8), then I pressed the button circled red; it proceeded as required, EXCEPT not all inspirations got turned on (they should have).
![image](https://user-images.githubusercontent.com/33267937/87001577-b77bc180-c1af-11ea-986d-d986bd0eb36e.png)

The other way round is also wrong (snapshot below): with not all fragments turned on, I clicked the down button (circled red): it proceeded correctly, except now it should have turned OFF all inspirations for that next compound (it didn't turn off anything).
![image](https://user-images.githubusercontent.com/33267937/87001650-d8dcad80-c1af-11ea-93d9-bece88d0e0ce.png)

**Comment by reskyner** [comment]
If inspirations are on in the RHS, when moving to the next compound, all inspirations for previous molecule should be turned off, and all inspirations for new molecule should be turned on 

SETTINGS: Make all settings applied to inspirations happen for all inspirations (?) @phraenquex

**Comment by ak-m2ms** [comment]
Analysis of Point: (Ideally) if any display settings were customised for either protein, ligand or contacts, then remember those too.

- Remember customised display control settings from previous compound/hit to moved compound/hit.

After code analysis I suppose there is option for quite simply solution:
- while moving compound/hit (after pressing arrow up/down) find object (ligand, protein...) of previous compound/hit in existing collection and insert representation (settings) of that to new moved object
- next modification: add option to insert "previous" representation when moved object (ligand, protein..) is added
- and apply this to LHS, RHS

**Comment by boriskovar-m2ms** [comment]
> 
> 
> Analysis of Point: (Ideally) if any display settings were customised for either protein, ligand or contacts, then remember those too.
> 
>     * Remember customised display control settings from previous compound/hit to moved compound/hit.
> 
> 
> After code analysis I suppose there is option for quite simply solution:
> 
>     * while moving compound/hit (after pressing arrow up/down) find object (ligand, protein...) of previous compound/hit in existing collection and insert representation (settings) of that to new moved object
> 
>     * next modification: add option to insert "previous" representation when moved object (ligand, protein..) is added
> 
>     * and apply this to LHS, RHS

I agree with the solution, but also please remove any custom representation settings from previous compound/molecule and reset them to default values.

**Comment by phraenquex** [comment]
Yes, agreed both with solution and Boris' qualification.

**Comment by boriskovar-m2ms** [comment]
RHS - if I turn on dialog for inspirations and select nothing in the inspirations then when I move down on the next molecule the contents of the modal window are the same. We need to load the inspirations for this next molecule also in this case. Also when switching to another (previous) compound please move the dialog with inspirations (F) or cross-references (X) to appropriate molecule because now it's not clear at what I'm looking at. Please also investigate possibility to adjust the vertical scroll-bar position for RHS and LHS when iterating through the molecules or compounds.
![image](https://user-images.githubusercontent.com/16878338/96444464-985ce900-120e-11eb-8c3b-a6fdcaf202e6.png)
![image](https://user-images.githubusercontent.com/16878338/96444698-030e2480-120f-11eb-9212-0e0e43339228.png)

Also when I open cross-references the content of the window always remains same and it needs to be updated (and moved) based on what compound we are at while iterating through them. 
![image](https://user-images.githubusercontent.com/16878338/96445026-8af42e80-120f-11eb-8f86-4a7504caea6a.png)
![image](https://user-images.githubusercontent.com/16878338/96445186-c55dcb80-120f-11eb-8358-e1256f93f0a4.png)

**Comment by teriraj** [comment]
Point 1:
RHS - I have  L turned on
LHS - I have L turned on also
Behaviour:
When I click to arrow on RHS, compound on LHS is turned off.
However:
RHS - I have  L turned on
LHS - I have L turned on also
Behaviour:
When I click to arrow on LHS, compound on RHS is still turned on. 

Point 2: 
Consider whether the behavior is correct.
If I turn on modal window F (inspiration), then turn on window X (cross reference), the windows are stacked on top of each other.
If I click F again, X is still displayed in the modal window. I have to remove the X modal window (close with a cross) before I see the F modal window.

**Comment by teriraj** [comment]
Point 3: 
Selected tab:
No action is happened when you click the arrows.
![image](https://user-images.githubusercontent.com/65952633/96718846-e4439580-13a8-11eb-8df7-2b3d556108c0.png)

**Comment by boriskovar-m2ms** [comment]
Point 2: I think this is fine as long as the window says whether it's inspirations or cross-reference dialog then it's fine.

**Comment by ak-m2ms** [comment]
Point 2 and 3 fixed.

**Comment by phraenquex** [comment]
Point 1:  this behaviour is fine.  Whatever was turned on the RHS can stay on when clicking arrows on the LHS.  

(Explanation:  you care about the LHS compounds when stepping through RHS cmpounds;  but you don't care about the RHS compounds when stepping through LHS compounds.)

Also:  the LHS arrows are not in fact a priority - so if they're causing trouble, then we should remove them.

**Comment by teriraj** [comment]
- [ ] Point 1: As Rachel said - this behaviour is fine. 
- [ ] Point 3: Selected tab works correctly

Some other problem:
After click "X" button error is displayed.

**Comment by phraenquex** [comment]
If X can't be easily fixed (i.e. <30min), please disable it - it's not currently important.  I am VERY anxious to have this in staging, today.

**Comment by boriskovar-m2ms** [comment]
The X button was disabled and issue is tracked in #444

**Comment by phraenquex** [comment]
This funcitonality will be hard to assess fully until performance is improved - so linked to #426.

**Comment by phraenquex** [comment]
Also, #445 is a remainig bug.

**Comment by teriraj** [comment]
Regression test was performed, found bugs are tracked in task no. #495, #498.
